great_expectations/datasource/data_connector/inferred_asset_azure_data_connector.py

```diff
@@ -0,0 +1,173 @@
+import logging
+import os
+import re
+from typing import List, Optional
+
+try:
+    from azure.storage.blob import BlobServiceClient
+except ImportError:
+    BlobServiceClient = None
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.core.batch import BatchDefinition
+from great_expectations.core.batch_spec import AzureBatchSpec, PathBatchSpec
+from great_expectations.datasource.data_connector.inferred_asset_file_path_data_connector import (
+    InferredAssetFilePathDataConnector,
+)
+from great_expectations.datasource.data_connector.util import list_azure_keys
+from great_expectations.execution_engine import (
+    ExecutionEngine,
+    PandasExecutionEngine,
+    SparkDFExecutionEngine,
+)
+
+logger = logging.getLogger(__name__)
```

Comment:
Could we please move this try/except below the logger and add a logger.debug statement (akin to what we have the GCS connectors)?

Reply:
@cdkini Nice catch -- done (in next commit).